### PR TITLE
JSON API: Update option endpoint jetpack only

### PR DESCRIPTION
**Pull request by lezama**
Alternative to https://github.com/Automattic/jetpack/pull/2901

### CODE REVIEW

**Comment by roccotripaldi** [comment]
This is looking good - I think we just need to expand the whitelist.

File: json-endpoints/jetpack/class.wpcom-json-api-get-option-endpoint.php [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+			return new WP_Error( 'option_name_not_in_whitelist', __( 'You must specify a whitelisted option_name', 'jetpack' ) );
+		}
+		return true;
+	}
+
+	function get_readable_options( $site_option = false ) {
+		if ( $site_option ) {
+			/**
+			 * Filter the list of site options that are readable from the API.
+			 *
```
Comment (jeherve) [inline json-endpoints/jetpack/class.wpcom-json-api-get-option-endpoint.php]: Could you add the module name as well, like here?
https://github.com/Automattic/jetpack/blob/master/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php#L36

Thanks!

File: json-endpoints/jetpack/class.wpcom-json-api-get-option-endpoint.php [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+			'widget_categories',
+			'widget_rss',
+			'widget_text',
+		);
+		$jetpack_compact_options = Jetpack_Options::get_option_names();
+		$jetpack_non_compact_options = Jetpack_Options::get_option_names( 'non_compact' );
+		$whitelist = array_merge( $readable_wordpress_options, $jetpack_compact_options, $jetpack_non_compact_options );
+		/**
+		 * Filter the list of options that are readable from the API.
+		 *
```
Comment (jeherve) [inline json-endpoints/jetpack/class.wpcom-json-api-get-option-endpoint.php]: The module name is missing here as well.

File: json-endpoints/jetpack/class.wpcom-json-api-get-option-endpoint.php [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+		 * @param array The default list of options includes most WordPress settings and Jetpack compact and non-compact options.
+		 */
+		return apply_filters( 'jetpack_api_readable_options', $whitelist );
+	}
+
+	function get_writable_options( $site_option = false ) {
+		if ( $site_option ) {
+			/**
+			 * Filter the list of site options that are writable via the API.
+			 *
```
Comment (jeherve) [inline json-endpoints/jetpack/class.wpcom-json-api-get-option-endpoint.php]: Also missing the module name

File: json-endpoints/jetpack/class.wpcom-json-api-get-option-endpoint.php [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+			'use_smilies',
+			'use_trackback',
+			'users_can_register',
+			'widget_categories',
+			'widget_rss',
+			'widget_text',
+		);
+		/**
+		 * Filter the list of options that are writable via the API.
+		 *
```
Comment (jeherve) [inline json-endpoints/jetpack/class.wpcom-json-api-get-option-endpoint.php]: Missing module name.

**Comment by roccotripaldi** [comment]
I've tested this, and it works as expected. I was able read and write only from whitelisted options, and I was able to filter the whitelist through a helper function:
```
function filter_json_api_readable_options( $options ) {
	$options[] = 'my_special_option';
	return $options;
}

add_filter( 'jetpack_api_readable_options', 'filter_json_api_readable_options' );
```

**Pull request merged**